keps/sig-scheduling/2891-simplified-config/README.md

```diff
@@ -0,0 +1,861 @@
+reference to any plugins that are only enabled through MultiPoint later when we 
+try to unroll them to their individual extension points.
+
+```go
+func (f *frameworkImpl) pluginsNeeded(plugins *config.Plugins) map[string]config.Plugin {
+	pgMap := make(map[string]config.Plugin)
+
+	if plugins == nil {
+		return pgMap
+	}
+
+	find := func(pgs *config.PluginSet) {
+		for _, pg := range pgs.Enabled {
+			pgMap[pg.Name] = pg
+		}
+	}
+	for _, e := range f.getExtensionPoints(plugins) {
+		find(e.plugins)
+	}
+    
+    // Now parse MultiPoint as well to store init functions
+    find(&plugins.MultiPoint)
+	return pgMap
+}
+```
+
+Either before or after [regular extension point plugin initialization](https://github.com/kubernetes/kubernetes/blob/dc079acc2be603fbc2065e2748c79bb5e2c3453d/pkg/scheduler/framework/runtime/framework.go#L349-L353) attempt 
+to initialize all `MultiPoint` plugins as all extension points (this is the section of 
+code that utilizes the factory function stored in the `frameworkImpl` struct)
+
+```go
+// initialize regular extension points (current behavior)
+for _, e := range f.getExtensionPoints(profile.Plugins) {
+  if err := updatePluginList(e.slicePtr, *e.plugins, pluginsMap); err != nil {
+    return nil, err
+  }
+}
+
+// initialize MultiPoint plugins
+for _, e := range f.getExtensionPoints(profile.Plugins) {
+  plugins := reflect.ValueOf(e.slicePtr).Elem()
+  pluginType := plugins.Type().Elem()
+  // build set of plugins already registered via normal extension points
+  // to check double registration
+  set := sets.NewString()
+  for _, plugin := range *e.plugins {
+    set.Insert(plugin.Name)
+  }
+  
+  for _, ep := range profile.Plugins.MultiPoint.Enabled {
+    pg, ok := pluginsMap[ep.Name]
+    if !reflect.TypeOf(pg).Implements(pluginType) {
+      continue
+    }
+    
+    if !ok {
+      return fmt.Errorf("%s %q does not exist", pluginType.Name(), ep.Name)
+    }
+    
+    if set.Has(ep.Name) {
+      return fmt.Errorf("plugin %q already registered as %q", ep.Name, pluginType.Name())
+    }
+    
+    set.Insert(ep.Name)
+    
+    newPlugins := reflect.Append(plugins, reflect.ValueOf(pg))
+    plugins.Set(newPlugins)
+  }
+}
+```
+
+This code is similar to the existing [`updatePluginList`](https://github.com/kubernetes/kubernetes/blob/dc079acc2be603fbc2065e2748c79bb5e2c3453d/pkg/scheduler/framework/runtime/framework.go#L418)
+function except for 2 notable differences:
+
+1. It does not return an error if a plugin doesn't implement an extension point, it just continues
+2. It is only looping through the MultiPoint config field for each internal extension point
+
+With this implementation, there is no option for plugins or users to define a subset of extension 
+points to enable or disable should we deprecate the existing regular extension points. This is because 
+the plugins themselves do not have control over when their functions are called, and we have cast them 
```

Comment:
That is actually better, we want the config api to be narrow and concise.

Reply:
just explaining why this means users will not have subset control if the old configuration is ever removed